### PR TITLE
Fix combiner glitch

### LINKED ISSUE #631: Clients shouldn't be colored after the project
Currently clients (Toggl; Teamweek) are colored after the project though they shouldn't be. Clients should always be #a3a3a3
Note: tasks on the other hand should be colored after the project

screen shot 2017-09-26 at 08 21 48

### PR DESCRIPTION
**Pull request by willsb**
- The binding context might run on a background thread for :rocket: 
- We were using functions that required running on the UI thread
- :boom: 

Fix:

I now cache the little dot and require the font size when constructing the combiner. This way stuff doesn’t break.

This also now requires the client color, since this was broken previously.

Squash whenever

Collaterally closes #631

### CODE REVIEW

File: Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs [hunk 1]
```diff
@@ -6,91 +6,89 @@
...
-using System.Globalization;
 using MvvmCross.Platform.UI;
-using CoreGraphics;
-using MvvmCross.Plugins.Color;
 using Toggl.Foundation.MvvmCross.Helper;
 using MvvmCross.Plugins.Color.iOS;
 using Toggl.Daneel.Extensions;
+using MvvmCross.Platform;
+using MvvmCross.Platform.Core;
+using System.Threading.Tasks;
```
Comment (arturs105) [inline Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs]: 10, 13, 14 and 15 🙅‍♂️

File: Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs [hunk 2]
```diff
@@ -6,91 +6,89 @@
...
 
 namespace Toggl.Daneel.Combiners
 {
     public sealed class ProjectTaskClientValueCombiner : MvxFormatValueCombiner
     {
         private const string projectDotImageResource = "icProjectDot";
+        private readonly UIColor clientColor;
+        private readonly NSMutableAttributedString cachedDotString;
+
+        public ProjectTaskClientValueCombiner(nfloat fontHeight, UIColor clientColor)
```
Comment (arturs105) [inline Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs]: Check the color for null?

File: Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs [hunk 3]
```diff
@@ -6,91 +6,89 @@
...
 
-        private static NSAttributedString tryAddColorToClient(string text, string client)
-        {
+            var text = builder.ToString();
+
             var result = new NSMutableAttributedString(text);
-            if (string.IsNullOrEmpty(client))
-                return result;
+            var clientIndex = text.Length - client.Length;
+            var projectNameRange = new NSRange(0, clientIndex);
```
Comment (arturs105) [inline Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs]: Doesn't this also include the task, since it's "Project: Task Client"?

**Comment by arturs105** [comment]
Also looks like this fixes #631. But at the same time breaks the project/task/client row in the edit view  - project shouldn't be colored. Maybe we should add a `bool` to indicate, whether the project text needs to be colored.

File: Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs (hunk 3 above)
Comment (willsb) [reply Toggl.Daneel/Combiners/ProjectTaskClientValueCombiner.cs]: Text includes the task as well

**Pull request merged**